src/lib/actions.js

```diff
@@ -182,3 +183,41 @@ export const setUserAcceptsCookies = store.action(() => {
     // overwrite it.
   };
 });
+
+// TODO(devnook): Extract isSupportedLocale to a common format.
+// Temp validation.
+function isValidLocale(locale) {
+  return ['en', 'pl'].indexOf(locale) > -1;
+}
+
+export const checkUserPreferredLocale = store.action(
+  ({userPreferredLocale}) => {
+    if (!userPreferredLocale) {
+      const cookie = cookies.get('preferred_lang');
+      if (isValidLocale(cookie)) {
+        return {
+          userPreferredLocale: cookie,
+        };
+      }
+    }
+    return {userPreferredLocale};
+  },
+);
+
+export const setLocale = store.action((state, preferredLocale) => {
+  if (!isValidLocale(preferredLocale)) {
+    console.warn('Invalid locale: ', preferredLocale);
+    return;
+  }
+  const options = {
+    expires: 365,
```

Comment:
I'm not a cookie expert, but would we want this to be some date far in the future, so it essentially never expires?